pandas/core/groupby/groupby.py

```diff
@@ -810,7 +812,18 @@ def __iter__(self) -> Iterator[tuple[Hashable, NDFrameT]]:
         for each group
         """
         keys = self.keys
+        level = self.level
         result = self.grouper.get_iterator(self._selected_obj, axis=self.axis)
+        if isinstance(level, list) and len(level) == 1:
```

Comment:
What happens for list-like objects that aren't subclasses of list? E.g. tuples, np.ndarray?

Reply:
is_list_like makes sense to me.

Reply:
Sounds good, I'll get it done tomorrow!(unfortunate time difference)